labgrid/driver/sshdriver.py

```diff
@@ -82,10 +83,8 @@ def _start_own_master_once(self, timeout):
         control = os.path.join(
             self.tmpdir, 'control-{}'.format(self.networkservice.address)
         )
-        # use sshpass if we have a password
-        args = ["sshpass", "-e"] if self.networkservice.password else []
 
-        args += ["ssh", "-f", *self.ssh_prefix, "-x", "-o", "ConnectTimeout={}".format(timeout),
+        args = ["ssh", "-f", *self.ssh_prefix, "-x", "-o", "ConnectTimeout={}".format(timeout),
                  "-o", "ControlPersist=300", "-o",
                  "UserKnownHostsFile=/dev/null", "-o", "StrictHostKeyChecking=no",
                  "-o", "ServerAliveInterval=15", "-MN", "-S", control.replace('%', '%%'), "-p",
```

Comment:
I just realized it, in line 90, instead of "-S", shouldn't it be "-o", "ControlPath=" + control.replace('%', '%%'), because a new socket needs to be created. The -S is for reusing, isn't it? I think the ControlMaster variable needs to be also set properly. Oh man, I am such a noob with openssh :)

Reply:
To me the description in `man ssh` and `man ssh_config` read like `-S` and `-o ControlPath=` are equivalent. The only way to find out for sure is testing.